src/effects/VST3/internal/x11/RunLoop.h

```diff
@@ -0,0 +1,66 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file RunLoop.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#pragma once
+
+#include <unordered_map>
+#include <gtk/gtk.h>
+#include <gdk/gdkx.h>
+#include <gdk/gdk.h>
+
+#include <pluginterfaces/gui/iplugview.h>
+
+namespace internal
+{
+    
+namespace x11
+{
+
+class RunLoop : public Steinberg::Linux::IRunLoop
+{
+   struct EventData
+   {
+      GIOChannel* channel;
+      guint id;
+   };
+
+   std::unordered_map<Steinberg::Linux::ITimerHandler*, guint> mTimerHandlers;
```

Comment:
Similarly the data type of this map might be instead a structure that calls `g_source_remove` in its destructor